dashboard/app/models/pd/pre_workshop_survey.rb

```diff
@@ -48,6 +59,6 @@ def lessons
   end
 
   def units_and_lessons
-    @units_and_lessons ||= workshop.try :pre_survey_units_and_lessons
+    @units_and_lessons ||= Pd::PreWorkshopSurvey.units_and_lessons(workshop)
```

Comment:
is there a specific reason for this caching?

Reply:
The [units_and_lessons](https://github.com/code-dot-org/code-dot-org/blob/8fc1f35d42eb7cc5d48a2558de3b4df98c884dba/dashboard/app/models/pd/workshop.rb#L613-L624) method performs several queries, and it may be called twice here to construct `units` and `lessons` above, so I figured I'd optimize. That said, perhaps the optimization is overzealous. I'm happy to remove it for simpler / more readable code.